src/navigation/mainStackNavigation.js

```diff
@@ -100,6 +110,9 @@ const MainStackNavigation = ( ): React.Node => (
         <Stack.Screen
           name="PhotoGallery"
           component={PhotoGalleryWithPermission}
+          options={{
+            headerTitle: photoGalleryHeaderTitle
```

Comment:
I personally prefer when the header options are set in the screen component itself. So that the main navigation files are only about the logic of navigation. Then also if I look at a screen component I can see all of its layout including the header (e.g., including translated strings). So, if I look at the phone all of the screen I see I know is in one file (or subcomponents of course).

It was a bit easier when using class components. Then the header setting could be on top of the file, and always in the same location in the code. Now with functional components it would be in a useEffect hook in between the other hooks.

Reply:
I guess I don't have a strong preference on this, but I kind of like that the headers live here. Since we're trusting react navigation to render the headers in a native way, it makes sense to me that they would be set in navigation, rather than where we're rendering the screen component.

I definitely don't love the idea of putting these in `useEffect` hooks except for the ones where we need an [interaction between the header and the screen component](https://reactnavigation.org/docs/header-buttons/#header-interaction-with-its-screen-component).

Reply:
Definitely benefits to both approaches, but I'm inclined to err on the side of setting this config in the `Screen` definition and not the component because that seems to be what React Navigation folks do in their documentation. Let's at least try that for a while and if it's really annoying we can try something else.